src/devtools/overlay-helpers.js

```diff
@@ -0,0 +1,125 @@
+const overlayDivClassName = `single-spa_overlay--div`;
+const hexRegex = /^#[A-Fa-f0-9]{6}$/g
+
+export function setOverlaysOnApp(app) {
+  const { options, selectors } = getSelectorsAndOptions(app)
+  applyOverlays(selectors, options, app.name)
+}
+
+function applyOverlays(selectors, options, appName) {
+  selectors.forEach((selector) => {
+    createOverlayWithText(selector, options, appName)
+  })
+}
+
+export function removeOverlaysFromApp(app) {
+  const { options, selectors } = getSelectorsAndOptions(app)
+  removeOverlayFromSelectors(selectors)
+}
+
+function deleteNode(node) {
+  node && node.remove && node.remove()
+}
+
+function getSelectorsAndOptions(app) {
+  return {
+    selectors: app.devtools.overlays.selectors.map((selector) => {
+      if (document.querySelector(selector)) {
+        return selector
+      } else {
+        return undefined
+      }
+    }).filter(selection => selection),
+    options: app.devtools.overlays.options || {}
+  }
+}
+
+function removeOverlayFromSelectors(selectors) {
+  selectors.forEach(selector => {
+    removeOverlay(selector)
+  })
+}
+
+function removeOverlay(selector) {
+  const element = document.querySelector(selector)
+  if (!element) {
+    return null
+  }
+  const existingOverlayDiv = element.querySelector(`.${overlayDivClassName}`)
+  deleteNode(existingOverlayDiv)
+}
+
+function createOverlayWithText(selector, options, appName) {
```

Comment:
Mostly a copy-paste of code from single-spa-canopy. There are some slight differences though.